whisper/cli/src/config/logger.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+use fern::colors::{Color, ColoredLevelConfig};
+use rlog::{LevelFilter};
+use std::io;
+
+fn setup_logger(verbosity: u32, logging_to_file: bool) -> Result<(), fern::InitError> {
+  let colors_line = ColoredLevelConfig::new()
+    .error(Color::Red)
+    .warn(Color::Yellow)
+    .info(Color::BrightBlack)
+    .debug(Color::BrightBlack)
+    .trace(Color::BrightBlack);
+
+  let colors_level = colors_line.clone()
+    .info(Color::Green);
+
+  let mut base_config = fern::Dispatch::new();
+
+  base_config = match verbosity {
+    0 => {
```

Comment:
Use `const` for these? Or strings. Not sure why this takes a `u32` tbh.

Reply:
I've created a separate branch and tried to change it to string in this commit https://github.com/paritytech/parity-ethereum/commit/6ba4ee4556a6f9689cc79b20d650e6b51b671495

However I get the following error, and I do not understand why it doesn't work, as I've used this technique to before with enums:
```
$ cargo build -p whisper-cli
error: expected one of `::`, `=>`, `if`, or `|`, found `.`
  --> whisper/cli/src/config/logger.rs:58:18
   |
58 |         LogLevel::Error.value() => {
   |                        ^ expected one of `::`, `=>`, `if`, or `|` here

warning: unused import: `LevelFilter`
  --> whisper/cli/src/config/logger.rs:18:12
   |
18 | use rlog::{LevelFilter};
   |            ^^^^^^^^^^^
   |
   = note: #[warn(unused_imports)] on by default

warning: unreachable expression
  --> whisper/cli/src/config/logger.rs:57:2
   |
57 |       base_config = match verbosity {
   |  _____^
58 | |         LogLevel::Error.value() => {
59 | |             base_config
60 | |             .level(LevelFilter::Error)
...  |
87 | |         },
88 | |     };
   | |_____^
   |
   = note: #[warn(unreachable_code)] on by default

error: aborting due to previous error
```

Reply:
Note that once I figure out how to get this to compile, I'll have to apply the same changes to the evmbin CLI PR https://github.com/paritytech/parity-ethereum/pull/10761/files